pkg/satellite/validation.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellite
+
+import (
+	"regexp"
+	"strings"
+)
+
+// ErrMissingField is used to indicate that field of entity is missing
+type ErrMissingField string
+
+// ErrInvalidField is used to indicate that field of entity has invalid value
+type ErrInvalidField string
+
+// Error is a method for generating error an error for missing field
+func (e ErrMissingField) Error() string {
+	return string(e) + " is required"
+}
+
+// Error is a method for generating error an error for invalid field value
+func (e ErrInvalidField) Error() string {
+	return string(e) + " is invalid"
+}
+
+// IsValid is a method for validating UserInfo entity
+func (userInfo *UserInfo) IsValid() error {
+	if userInfo.FirstName == "" {
+		return ErrMissingField("first name")
```

Comment:
```
var errs []error
if userInfo.FirstName == "" {
    errs = append(errs, ErrMissingField("first name"))
}
if userInfo.Email == "" {
    errs = append(errs, ErrMissingField("email"))
}
if !checkEmailAddressSyntax(userInfo.Email) {
    errs = append(errs, ErrInvalidField("email"))
}
if userInfo.Password == "" {
    errs = append(errs, ErrMissingField("password"))
} else if len(userInfo.Password) < 6 {
    errs = append(errs, ErrInvalidField("password"))
}
return utils.CombineErrors(errs...)
```

This way you get all the mistakes, not just the first. It's annoying to fix things one-by-one, rather than multiple at the same time.